core/src/main/java/org/bitcoinj/core/TransactionConfidence.java

```diff
@@ -527,6 +527,16 @@ public synchronized void setSource(Source source) {
         this.source = source;
     }
 
+    /**
+     * Called when we receive a transaction from the network. It's possible we may have already seen this locally
+     * so we don't want to override a setting of Source.SELF, but if source was not set, we should set it.
+     */
+    public synchronized void setSourceToNetworkMaybe() {
```

Comment:
I renamed the method to `maybeSetSourceToNetwork()` and annotated it with `@VisibleForTesting` (unfortunately, we can't make it package-private because it is used by `o.b.testing.FakeTxBuilder`.